### PR TITLE
Maxbader jade devel pr #365 with merges to recent changes

### PR DESCRIPTION
**Pull request by hsu**
comments from #365:
- fixed the gazebo_ros_diff_drive for jade and gazebo5.
- There was a problem because of some deprecated functions.
- I also updated the test environment for some quick testing.

To test, run:

```
roslaunch gazebo_plugins diff_drive.launch
```

or

```
roslaunch gazebo_plugins diff_drive_multi_robot.launch
```

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_diff_drive.cpp [hunk 1]
```diff
@@ -284,14 +278,18 @@ void GazeboRosDiffDrive::UpdateChild()
...
 #if GAZEBO_MAJOR_VERSION > 2
-            joints_[LEFT]->SetParam ( "vel", 0, wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
-            joints_[RIGHT]->SetParam ( "vel", 0, wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
+        joints_[LEFT]->SetParam ( "vel", 0, wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
+        joints_[RIGHT]->SetParam ( "vel", 0, wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
 #else
-            joints_[LEFT]->SetVelocity ( 0,wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
-            joints_[RIGHT]->SetVelocity ( 0,wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
+        joints_[LEFT]->SetVelocity ( 0,wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
+        joints_[RIGHT]->SetVelocity ( 0,wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
```
Comment (scpeters) [inline gazebo_plugins/src/gazebo_ros_diff_drive.cpp:290]: should these be setting `wheel_speed_instr_` or `wheel_applied_vel`?

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_diff_drive.h [hunk 2]
```diff
@@ -107,6 +107,7 @@ namespace gazebo {
       double wheel_speed_instr_[2];
 
       std::vector<physics::JointPtr> joints_;
+      double wheel_applied_vel[2];
```
Comment (scpeters) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_diff_drive.h:110]: this should have a trailing `_` in the name

**Comment by j-rivero** [comment]
run test please

**Comment by maxbader** [comment]
This pull request as well as the maxbader-jade-devel branch can be deleted

**Pull request closed without merging**